extensions/azure/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetQueryBuilder.java

```diff
@@ -0,0 +1,37 @@
+package org.eclipse.dataspaceconnector.assetindex.azure;
+
+import org.eclipse.dataspaceconnector.assetindex.azure.model.AssetDocument;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.asset.Criterion;
+
+import java.util.StringJoiner;
+
+public class CosmosAssetQueryBuilder {
+
+    private static final String NAME = AssetDocument.class.getSimpleName();
+
+    private CosmosAssetQueryBuilder() {
+    }
+
+    public static String from(AssetSelectorExpression expression) {
+        String query = "SELECT * FROM " + NAME;
+        if (expression != AssetSelectorExpression.SELECT_ALL) {
+            query += buildWhereClause(expression);
+        }
+        return query;
+    }
+
+    private static String buildWhereClause(AssetSelectorExpression expression) {
+        StringJoiner joiner = new StringJoiner(" AND ");
+        expression.getCriteria().stream()
+                .map(CosmosAssetQueryBuilder::toSql)
+                .forEach(joiner::add);
+        return " WHERE " + joiner.toString();
+    }
+
+    private static String toSql(Criterion criterion) {
+        return "( AssetDocument.wrappedInstance.properties." + criterion.getOperandLeft() +
+                " " + criterion.getOperator() + " " +
+                criterion.getOperandRight() + " )";
+    }
```

Comment:
It's possible do an SQL injection here by manipulating the expression. E.g.
```java
 var expression = AssetSelectorExpression.Builder.newInstance()
                .constraint("name", "=", "hello\") OR (\"1\"=\"1")
                .build();
```

I'd suggest using the classes from the Azure Cosmos Library, something like this
```java
ArrayList<SqlParameter> paramList = new ArrayList<SqlParameter>();
        paramList.add(new SqlParameter("@id", "AndersenFamily"));
        SqlQuerySpec querySpec = new SqlQuerySpec(
                "SELECT * FROM Families f WHERE (f.id = @id)",
                paramList);

        executeQueryWithQuerySpecPrintSingleResult(querySpec);
```
Example Code from
https://github.com/Azure-Samples/azure-cosmos-java-sql-api-samples/blob/main/src/main/java/com/azure/cosmos/examples/queries/sync/QueriesQuickstart.java#L392-L421

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

Reply:
To be honest I do not see any added value in using the cosmos library here, as it does not facilitate in any manner the construction of the SQL query.  
However, based on your comment, I searched for some existing libraries that would actually help in construction the SQL query, and I found out JOOQ, which appears to be the main one. 
I reimplemented the query builder by leveraging on this library which, in my opinion, makes the query construction much better.

Reply:
@bscholtes1A what's the license JOOQ is distributed under?

Reply:
Hi @jimmarino 
looking at the doc: https://www.jooq.org/doc/3.0/manual/copyright/ it seems it is dual-licensed:
- Apache software license 2.0 for use with Open source database
- commercial for use with commercial databases.
Not an expert but does not seem to be good then... 

Note that it was only Apache License v2 until version <= 3.1 o jOOQ. So maybe a downgrade would be enough?

Reply:
Dependency on JOOQ removed